tests/unit/test_expression_tree/test_parameter.py

```diff
@@ -22,22 +22,58 @@ def test_evaluate_for_shape(self):
 class TestFunctionParameter(unittest.TestCase):
     def test_function_parameter_init(self):
         var = pybamm.Variable("var")
-        func = pybamm.FunctionParameter("func", var)
+        func = pybamm.FunctionParameter("func", {"var": var})
         self.assertEqual(func.name, "func")
         self.assertEqual(func.children[0].id, var.id)
         self.assertEqual(func.domain, [])
         self.assertEqual(func.diff_variable, None)
 
     def test_function_parameter_diff(self):
         var = pybamm.Variable("var")
-        func = pybamm.FunctionParameter("a", var).diff(var)
+        func = pybamm.FunctionParameter("a", {"var": var}).diff(var)
         self.assertEqual(func.diff_variable, var)
 
     def test_evaluate_for_shape(self):
         a = pybamm.Parameter("a")
-        func = pybamm.FunctionParameter("func", 2 * a)
+        func = pybamm.FunctionParameter("func", {"2a": 2 * a})
         self.assertIsInstance(func.evaluate_for_shape(), numbers.Number)
 
+    def test_copy(self):
+        a = pybamm.Parameter("a")
+        func = pybamm.FunctionParameter("func", {"2a": 2 * a})
+
+        new_func = func.new_copy()
+        self.assertEqual(func._input_names, new_func._input_names)
+
+    def test_print_input_names(self):
+        var = pybamm.Variable("var")
+        func = pybamm.FunctionParameter("a", {"var": var})
+        func.input_names
```

Comment:
not sure what this test is supposed to do?

Reply:
It tests that printing input names works without raising an error. As above `func.input_names` will print the imput names